compiler/ilgen/IlBuilder.hpp

```diff
@@ -198,6 +198,9 @@ class IlBuilder : public TR::IlInjector
    TR::IlValue *IndexAt(TR::IlType *dt, TR::IlValue *base, TR::IlValue *index);
    TR::IlValue *AtomicAddWithOffset(TR::IlValue *baseAddress, TR::IlValue *offset, TR::IlValue *value);
    TR::IlValue *AtomicAdd(TR::IlValue *baseAddress, TR::IlValue * value);
+   void TransactionBegin(TR::IlBuilder **persistentFailureBuilder, TR::IlBuilder **transientFailureBuilder, TR::IlBuilder **fallThroughBuilder);
```

Comment:
Yes Mark, now I moved what was in TransactionEnd() to the end of fallThrough IlBuilder and change the service name to Transaction().